Feature: Option to disable activate-on-release behaviour of toolbar dropdown buttons.

PeterN opened the pull request:
## Motivation / Problem

A common problem for new users is they don't realise the main toolbar buttons bring up a dropdown menu as it disappears straight after clicking.

Therefore they cannot find how to change railtypes nor find a few more features probably.

## Description

This change introduces an option to change this behaviour (by default) so that clicking and releasing simply opens the dropdown menu.

This is probably game-breaking for existing users but if the default value maintains existing behaviour then it won't benefit new users at all.

## Limitations

<!--
Describe here
* Is the problem solved in all scenarios?
* Is this feature complete? Are there things that could be added in the future?
* Are there things that are intentionally left out?
* Do you know of a bug or corner case that does not work?
-->


## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

TrueBrain commented on src/lang/english.txt:
```diff
@@ -1568,6 +1568,9 @@ STR_CONFIG_SETTING_SCROLLMODE_LMB                               :Move map with L
 STR_CONFIG_SETTING_SMOOTH_SCROLLING_HELPTEXT                    :Control how the main view scrolls to a specific position when clicking on the smallmap or when issuing a command to scroll to a specific object on the map. If enabled, the viewport scrolls smoothly, if disabled it jumps directly to the targeted spot
 
+STR_CONFIG_SETTING_TOOLBAR_DROPDOWN_CLOSE                       :Toolbar dropdowns activate on release: {STRING2}
+STR_CONFIG_SETTING_TOOLBAR_DROPDOWN_CLOSE_HELPTEXT              :Control whether tthe main toolbar dropdowns activate their default function when released, or if they require an additional selection.
```
tthe? :D Drop the t please :) Just one, not both.

TrueBrain replied:
This sentence is really hard to read, and even after reading it 4 times, I am not exactly sure what it does :P Maybe dumb it down a bit?

TrueBrain reviewed (commented):
I like the idea of this PR :)

Does it still mean that if you click it once, it selects the default? Or do you always have to click twice?

PeterN committed:
Fix: Close dropdown menu when user clicks on a differenet window.

This did not happen if the clicked window is flagged as unfocusable (i.e. the main toolbar.)

PeterN committed:
Feature: Option to disable activate-on-release behaviour of toolbar dropdown buttons.

2TallTyler commented on src/lang/english.txt:
```diff
@@ -1690,6 +1690,9 @@ STR_CONFIG_SETTING_SCROLLMODE_LMB                               :Move map with L
 STR_CONFIG_SETTING_SMOOTH_SCROLLING                             :Smooth viewport scrolling: {STRING2}
 STR_CONFIG_SETTING_SMOOTH_SCROLLING_HELPTEXT                    :Control how the main view scrolls to a specific position when clicking on the smallmap or when issuing a command to scroll to a specific object on the map. If enabled, the viewport scrolls smoothly, if disabled it jumps directly to the targeted spot
 
+STR_CONFIG_SETTING_TOOLBAR_DROPDOWN_AUTOSELECT                  :Toolbar dropdowns automatically activate: {STRING2}
```
This is a hard thing to explain, so what about a different approach of not trying to fully explain it? Something like `Quick toolbar dropdowns` or even `Legacy toolbar dropdowns?

2TallTyler commented on src/lang/english.txt:
```diff
@@ -1690,6 +1690,9 @@ STR_CONFIG_SETTING_SCROLLMODE_LMB                               :Move map with L
 STR_CONFIG_SETTING_SMOOTH_SCROLLING_HELPTEXT                    :Control how the main view scrolls to a specific position when clicking on the smallmap or when issuing a command to scroll to a specific object on the map. If enabled, the viewport scrolls smoothly, if disabled it jumps directly to the targeted spot
 
+STR_CONFIG_SETTING_TOOLBAR_DROPDOWN_AUTOSELECT                  :Toolbar dropdowns automatically activate: {STRING2}
+STR_CONFIG_SETTING_TOOLBAR_DROPDOWN_AUTOSELECT_HELPTEXT         :Control whether the main toolbar dropdowns activate their default function when the mouse button is released. If not enabled then the dropdown menu will stay open so a selection can be made.
```
```suggestion
STR_CONFIG_SETTING_TOOLBAR_DROPDOWN_AUTOSELECT_HELPTEXT         :Enable legacy main toolbar behavior. If enabled, clicking on a main toolbar button will activate its default function while holding the mouse button will open the dropdown, selecting whichever options is hovered over when released. If disabled, clicking on a main toolbar button will open the dropdown menu without needing to hold the mouse button.
```

PeterN replied:
I think "legacy" is a bit strong, the current way is not being deprecated.

2TallTyler replied:
Traditional?

PeterN replied:
IMHO this text is too wordy, and is wrong. We can omit unnecessary details but if we include them it should be correct. (The dropdown menus are always displayed when clicking, they just close again after a brief time.)
